File: 1921-eliminate-maximum-number-of-monsters/1921-eliminate-maximum-number-of-monsters.py
class Solution:
    def eliminateMaximum(self, dist: List[int], speed: List[int]) -> int:
        n = len(dist)
        time = [dist[i]/speed[i] for i in range(n)]
        time.sort()
        
        curr = 0
        for i in range(n):
            if(curr>=time[i]):
                return i
            curr+=1
        return n

File: 1921-eliminate-maximum-number-of-monsters/1921-eliminate-maximum-number-of-monsters.java
class Solution {
    public int eliminateMaximum(int[] dist, int[] speed) {
        int n = dist.length;
        float[] time = new float[n];
        for(int i=0; i<n; ++i)
            time[i] = (float)dist[i]/(float)speed[i];
        
        Arrays.sort(time);
        int curr = 0;
        for(int i=0; i<n; ++i){
            if(curr++ >= time[i])
                return i;
        }
        return n;
    }
}